src/service/command/dice.ts

```diff
@@ -49,23 +49,37 @@ export class DiceCommand implements CommandResponder<typeof SCHEMA> {
   constructor(private readonly diceQueen: DiceQueen) {}
 
   async on(message: CommandMessage<typeof SCHEMA>): Promise<void> {
+    const regExState = /^(?<num>\d+)[dD](?<faces>\d+)$/;
     const [arg] = message.args.params;
 
-    if (arg.match(/^(?!a-c,e-z)*$/)) {
+    const matchResult = regExState.exec(arg);
+
+    if (matchResult == null) {
       await message.reply({
         title: 'コマンド形式エラー',
         description: '引数の形は`<num>d<num>`をとる必要があるよ。'
       });
       return;
     }
-    const [arg1, arg2] = arg.toLowerCase().split('d', 2);
-    const diceNum = parseInt(arg1);
-    const diceFaces = parseInt(arg2);
+    const arg1 = matchResult.groups?.num;
+    const arg2 = matchResult.groups?.faces;
+    const diceNum = parseInt(arg1, 10);
+    const diceFaces = parseInt(arg2, 10);
 
-    if (diceFaces * diceNum >= 2000) {
+    if (
+      !(
+        1 <= diceFaces &&
+        diceFaces <= 100 &&
+        1 <= diceNum &&
+        diceNum <= 20 &&
+        1 <= diceFaces * diceNum &&
+        diceFaces * diceNum <= 2000
```

Comment:
```suggestion
        diceNum <= 20
```
この条件なら掛けた値についての検査は不要かと